lib/src/_shared/hooks/usePickerState.ts

```diff
@@ -103,8 +108,9 @@ export function usePickerState<TInput, TDateValue>(
         wrapperVariant: WrapperVariant,
         selectionState: PickerSelectionState = 'partial'
       ) => {
-        setPickerDate(newDate);
-
+        if (!valueManager.areDaysEqual(utils, newDate, pickerDate)) {
+          setPickerDate(newDate);
+        }
```

Comment:
```suggestion
        if (!valueManager.areDaysEqual(utils, newDate, pickerDate)) {
          return;
        }

        setPickerDate(newDate);
```

Reply:
I might be missing something but wouldn't this run only if you select the same date?

Reply:
Yes, you are right but we still doesn't need to update the state. 
Another question – what for TimePicker and DateTimePicker? I think overall we need to add this check to the `Calendar` component.